MushroomObserver/mushroom-observer
# Fungarium record MyCoPortal link

Links to the fungarium's collection record on MyCoPortal.
- Delivers #2471, which is inspired by a requested from MO User pinonbistro;
- Adds a `mycoportal_db` integer column to the Herbarium table;
- Allows populating that column via the Herbarium form; 
- For MO Fungarium Records, for MO Fungaria with `mycoportal_db`, links to the fungarium's collection record (in MCP), **based on the Accession Number**:
  - on the MO Fungarium Record page; and
  - on the MO Observation page.

### Suggested Manual Test
- Enter Admin Mode;
- Edit an MO institutional Herbarium by filling in the `MyCoPortal Collection ID`;
Important:
  - I suggest the [Robert L. Gilbertson Mycological Herbarium](http://localhost:3000/herbaria/407) or [University of Tennessee ](http://localhost:3000/herbaria/66/edit) because many of their MO Herbarium Records have an accurate Accession Number.
  - To find the Collection ID, click the help icon (a question mark) to the right of `MyCoPortal Collection ID`
- Exit Admin Mode;
- Click "This fungarium contains _nnnn_  fungarium record(s).";
- Click on an Accession Number;
- Click the link which takes the form "_herbarium abbrev_ Collection Record".
Expected result:  Opens a new tab with the MyCoPortal collection record corresponding to that Accession Number.
(in many cases the result is "Your query did not return any results." because the MO Accession Number is inaccurate.)

### Notes
After deployment
- I plan to immediately fill in `MyCoPortal Collection ID` for some MO Fungaria;
- I plan to add a Article soon after deployment.
- I may directly contact some users who have many MO Fungarium Records to get feedback.
